core/src/avm2/regexp.rs

```diff
@@ -204,17 +209,68 @@ impl<'gc> RegExp<'gc> {
         ret
     }
 
-    pub fn replace(
+    // Implements string.replace(regex, replacement) where the replacement is
+    // a function.
+    pub fn replace_fn(
+        &mut self,
+        activation: &mut Activation<'_, 'gc, '_>,
+        text: AvmString<'gc>,
+        f: &FunctionObject<'gc>,
+    ) -> AvmString<'gc> {
+        self.replace_with_fn(activation, &text, |activation, txt, m| {
+            let mut args = std::iter::once(Some(&m.range))
+                .chain((m.captures.iter()).map(|x| x.as_ref()))
+                .map(|o| match o {
+                    Some(r) => Value::from(AvmString::new(
```

Comment:
nit: This can be (simpler in my opinion):
```rs
Some(r) => AvmString::new(
    activation.context.gc_context,
    &txt[r.start..r.end],
).into(),
```
The same applies for other instances of `Value::from()` as well.